src/main/java/chatbot/parser/Parser.java

```diff
@@ -0,0 +1,75 @@
+package chatbot.parser;
+
+import chatbot.commands.*;
+import chatbot.exceptions.DukeException;
+
+/**
+ * Represents the class that is responsible for parsing user inputs and generating the relevant commands
+ */
+public class Parser {
+    /**
+     * Checks if the user input is valid
+     * @param command User input to be validated
+     * @return Returns a boolean that indicates if the input is valid
+     * @throws DukeException If the user input is not valid
+     */
+    public static boolean isValid (String command) throws DukeException {
+        if(command.equals("list") || command.equals("bye")) {
+            return true;
+        } else if(command.split(" ")[0].equals("todo") && command.split(" ")[1].length() < 1) {
+            throw new DukeException("Error! Please include a description of the todo");
+        } else if(command.split(" ")[0].equals("deadline") && command.split(" ")[1].length() < 1
+                && !command.contains("/by")) {
+            throw new DukeException("Error! Input should have a description and date");
+        } else if(command.split(" ")[0].equals("event") && command.split(" ")[1].length() < 1
+                && !command.contains("/from") && !command.contains("/to")) {
+            throw new DukeException("Error! Input should have a description, start date and end date");
+        } else if(command.split(" ")[0].equals("mark") && command.split(" ")[1].length() < 1) {
+            throw new DukeException("Error! Please include a description of the task to be marked");
+        } else if(command.split(" ")[0].equals("unmark") && command.split(" ")[1].length() < 1) {
+            throw new DukeException("Error! Please include a description of the task to be unmarked");
+        } else {
+            return true;
+        }
+    }
```

Comment:
Maybe this could be delegated to the Command classes to figure out what kind of syntax they should accept?

Granted, implementation might get a bit messy, and this certainly works!